### PR TITLE
Transpose Rob's hackathon changes

### PR DESCRIPTION
**Pull request by YanRachel**
Tested using http server: https://issues.citrite.net/browse/CP-41838

### CODE REVIEW

**Review by jameshensmancitrix — APPROVED** [review]

**Review by snwoods — APPROVED** [review]

File: ocaml/xenopsd/c_stubs/xenctrlext_stubs.c [hunk 1]
```diff
@@ -441,18 +443,22 @@ CAMLprim value stub_xenctrlext_cputopoinfo(value xch)
 	CAMLreturn(result);
 }
 
-CAMLprim value stub_xenforeignmemory_open(value unit)
+CAMLprim value stub_xenforeignmemory_open(value logger)
```
Comment (edwintorok) [inline ocaml/xenopsd/c_stubs/xenctrlext_stubs.c:446]: This seems to undo some changes from master, was it rebased correctly?

File: xapi-xenopsd-xc.opam [hunk 2]
```diff
@@ -45,6 +45,7 @@ depends: [
   "xenctrl"
   "xenstore"
   "xenstore_transport"
+  "xentoollog"
```
Comment (andyhhp) [inline xapi-xenopsd-xc.opam:48]: Looks like another rebasing mistake.  This was specifically deleted recently due to it having been deleted upstream.

File: ocaml/xenopsd/c_stubs/xenctrlext_stubs.c (hunk 1 above)
Comment (edwintorok) [reply ocaml/xenopsd/c_stubs/xenctrlext_stubs.c:446]: See [this PR ](https://github.com/xapi-project/xen-api/pull/4905/files)

**Comment by robhoes** [comment]
Hmm... I did split my changes into logical commits for good reasons. But now they are squashed together? This makes it very hard to see whats actually going on and what changes you have made yourself.

**Review by psafont — CHANGES_REQUESTED** [review]
This commit introduces a lot of code, please explain in the commit message what is being added and why.

Xapi is an open-source project and not only people in xenserver need to have context of the changes

**Pull request closed without merging**